### ISSUE TITLE
False Positive "no-member" with pandas `DatetimeIndex`

### ISSUE THREAD

**Issue by sam-s** [pylint-dev/pylint#10796]
### Bug description

pylint thinks that [`pd.DatetimeIndex.date`](https://pandas.pydata.org/docs/reference/api/pandas.DatetimeIndex.date.html) does not exist:

```python
"https://github.com/pylint-dev/pylint/issues/10796"
import pandas as pd
print(pd.date_range(start=pd.Timestamp("now"),periods=3).date)
```

### Command used

```shell
pylint pylint-10796.py
```

### Pylint output

```python
************* Module pylint-10796
pylint-10796.py:3:6: E1101: Instance of 'DatetimeIndex' has no 'date' member (no-member)
```

### Expected behavior

no diagnostics

### Pylint version

```shell
pylint 4.0.4
astroid 4.0.2
Python 3.13.7 (main, Nov 24 2025, 20:51:28) [GCC 15.2.0]
```

### OS / Environment

ubuntu

### Additional dependencies

```python
pandas==2.3.3
```

**Comment by sam-s** [comment]
maybe related: https://github.com/pylint-dev/pylint/issues/10166

**Comment by MehrazRumman** [comment]
Trying to fix this issue !

**Comment by MehrazRumman** [comment]
@Pierre-Sassoulas I was looking at the [typecheck.py](https://github.com/pylint-dev/pylint/blob/main/pylint/checkers/typecheck.py) . I couldn’t find a optimal solution for this , can you give me any suggestion?

**Comment by Pierre-Sassoulas** [comment]
Sorry I probably can't in a reasonable time as I'm not using pandas a lot currently.

**Comment by DanielNoord** [comment]
This probably requires a "brain" for `astroid` rather than being fixed in `pylint`. There are many examples in the `astroid` codebase but it might be quite an undertaking to add a completely new one.
